### DISCUSSION TITLE
Collapsible sidebar with Alpine shows itself for split second on page load

### DISCUSSION THREAD

**Discussion by stackerito** [alpinejs/alpine#1864] [1. Help] [answered]
I have a weird bug where the sidebar with```<aside>``` tag shows itself for a split second when the page loads. It should be hidden unless you click the Hamburger. But it keeps showing up for a split second.
This is the code: https://codepen.io/Stackerito/pen/ExmQQJa

**Comment by KevinBatdorf** [comment, marked as answer]
Check out `x-cloak`

https://alpinejs.dev/directives/cloak

**Reply by stackerito** [reply]
Yep that worked, but I had to manually write ```x-cloak``` for multiple elements, not the best practice.. I hope they will fix this issue.. it feels like they made this ```x-cloak``` as a workaround for a bug instead of addressing the real issue

**Reply by alexej-d** [reply]
How should you fix that? HTML is rendered before Javascript kicks in. There is no other way to do that if you are not using a framework like React or Vue

**Reply by stackerito** [reply]
I thought they could find a solution like they do in React or Vue, but now that you specifically mentioned these two, I guess it's something deeper than I understand 😁

**Reply by SimoTod** [reply]
Vue has v-cloak to address the same problem in certain cases.

**Reply by stackerito** [reply]
@SimoTod thank you! Just saw it - good to know
